api/core/model_runtime/model_providers/yi/llm/llm.py

```diff
@@ -0,0 +1,988 @@
+import logging
+from collections.abc import Generator
+from typing import Optional, Union, cast
+
+import tiktoken
+from openai import OpenAI, Stream
+from openai.types import Completion
+from openai.types.chat import ChatCompletion, ChatCompletionChunk, ChatCompletionMessageToolCall
+from openai.types.chat.chat_completion_chunk import ChoiceDeltaFunctionCall, ChoiceDeltaToolCall
+from openai.types.chat.chat_completion_message import FunctionCall
+
+from core.model_runtime.callbacks.base_callback import Callback
+from core.model_runtime.entities.llm_entities import LLMMode, LLMResult, LLMResultChunk, LLMResultChunkDelta
+from core.model_runtime.entities.message_entities import (
+    AssistantPromptMessage,
+    ImagePromptMessageContent,
+    PromptMessage,
+    PromptMessageContentType,
+    PromptMessageTool,
+    SystemPromptMessage,
+    TextPromptMessageContent,
+    ToolPromptMessage,
+    UserPromptMessage,
+)
+from core.model_runtime.entities.model_entities import AIModelEntity, FetchFrom, I18nObject, ModelType, PriceConfig
+from core.model_runtime.errors.validate import CredentialsValidateFailedError
+from core.model_runtime.model_providers.__base.large_language_model import LargeLanguageModel
+from core.model_runtime.model_providers.openai._common import _CommonOpenAI
+
+logger = logging.getLogger(__name__)
+
+OPENAI_BLOCK_MODE_PROMPT = """You should always follow the instructions and output a valid {{block}} object.
+The structure of the {{block}} object you can found in the instructions, use {"answer": "$your_answer"} as the default structure
+if you are not sure about the structure.
+
+<instructions>
+{{instructions}}
+</instructions>
+"""
+
+class OpenAILargeLanguageModel(_CommonOpenAI, LargeLanguageModel):
```

Comment:
Maybe we can rename this class to `YiLargeLanguageModel` and make it inherited from `core.model_runtime.model_providers.openai_api_compatible.llm.OAIAPICompatLargeLanguageModel` so that everything will become easier.

Reply:
done. witty advice